content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,353 @@
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-directory.png")
+    ```
+
+
+
+### Add files and directories for a tutorial
+
+Now that the package structure is in place, it's time to add tutorial content. Again, the usethis package will do the dirty work for us to create the folders and files we need for a learnr tutorial within our package.
+
+1. **Create the learnr tutorial** and open it for editing when you run `usethis::use_tutorial("<tutorial-name>", "<Tutorial Title>")`:
+
+    ```{r, eval = FALSE}
+    usethis::use_tutorial("lesson1", "Your First Lesson", open = interactive())
+    ```
+
+    * The first argument is the filename for the learnr R Markdown file you are creating without the `.Rmd` extension.
+
+    *  The second argument is what you want the user-facing title of the tutorial to be.
+
+    * This creates an `inst/` directory with some subfolders. If you're not new to creating learnr tutorials, the folder structure here will look familiar to you -- there's nothing different about learnr tutorial directories inside and outside of packages.
+
+      You can open up `inst/` and its subsequent subfolders until you reach the `.Rmd` file you created:
+
+      ```
+      /inst/tutorials/<tutorial-name>/<tutorial-name.Rmd>
+      ```
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-inst.png")
+    ```
+
+1. **Edit the open `.Rmd` file** and write your tutorial as you normally would.
+
+1. **Repeat** for each additional tutorial you want to include in this package.
+
+
+
+### Build, install, and use your tutorial package
+
+Time to build the package and install it in your R system library.
+
+1. __Navigate to the *Build* pane__ of your RStudio IDE. Then __click *Install and Restart*__. 
+
+    * Alternatively, run:
+
+      ```{r, eval = FALSE}
+      devtools::install()
+      ```
+
+    * Believe it or not, you've now made a package with a learnr tutorial inside of it!
+
+    * You should test that your package has been properly created by loading your tutorial from your installed package:
+    
+1. __Navigate to the *Tutorial* pane__ of your RStudio IDE. All installed learnr tutorials in your R Library will be automatically indexed and displayed here. 
+
+    ```{r, echo = FALSE, out.width="90%"}
+    knitr::include_graphics("pkg-run-tutorial.png")
+    ```
+    
+    
+1. __Click Start Tutorial__ to open the tutorial from inside the tutorial pane (which can then be made bigger or popped out to a browser window) -- `r emo::ji("tada")` woot!
+  
+
+    * Alternatively, you can run your tutorial by using `learnr::run_tutorial("<tutorial-name>", "<package-name>")`.
+
+        ```{r, eval = FALSE}
+        learnr::run_tutorial("lesson1", "tutorialpackage")
+        ```
+
+    * The tutorial name here refers to the name of the `.Rmd` file, not the user-facing title.
+    * The live tutorial will open up in a local browser window.
+    
+
+### (Optionally) Add dependencies
```

Comment:
It might be nice to also show something like, "if your tutorial uses palmerpenguins use `use_package("palmerpenguins")` just so it's clear things used in exercises (i.e. learners will use) also go here.